src/Illuminate/Support/Collection.php

```diff
@@ -300,6 +300,27 @@ public function flip()
     }
 
     /**
+     * Determine if the callable is true for all items in the collection.
+     *
+     * @param  callable $callable
+     *
+     * @return bool
+     */
+    public function forAll(callable $callable)
+    {
+        $values = $this->values()->toArray();
+
+        while (! empty($values)) {
+            $item = array_shift($values);
+            if (! $callable($item)) {
+                return false;
+            }
+        }
+
+        return true;
+    }
```

Comment:
Any reason to not just use a `foreach` loop here?

```php
public function forAll(callable $callable)
{
    foreach ($this->items as $item) {
        if (! $callable($item) {
            return false;
        }
    }

    return true;
}
```

Could also use `contains`, but `foreach` won't unnecessarily check additional elements after finding something that returns false. Nevertheless the `contains` version is still sort of zonda so I'm obliged to share:

```php
public function forAll(callable $callable)
{
    return $this->contains(function ($key, $value) use ($callable) {
        return ! $callable($item);
    });
}
```